sfdx.js

```diff
@@ -0,0 +1,126 @@
+const util = require('util');
+const exec = util.promisify(require('child_process').exec);
+
+const {
+    createJWTFile,
+    deleteJWTFile,
+    JWT_KEY_FILE_DIRECTORY
+} = require('./jwtFile');
+
+const CLI = 'sfdx';
+const COMMANDS = {
+    AUTHENTICATE: 'auth:jwt:grant',
+    DEPLOY: 'force:source:deploy',
+};
+
+async function authenticate({ consumerKey, jwtKey, username, instanceUrl, sourceDirectory }) {
+    try {
+        await createJWTFile({
+            sourceDirectory,
+            jwtKey
+        });
+        await callAuthenticateCommand({
+            username,
+            consumerKey,
+            instanceUrl,
+            sourceDirectory
+        });
+    } finally {
+        await deleteJWTFile({
+            sourceDirectory
+        });
+    }
+}
+
+async function deployProject({
```

Comment:
Sure, but still this could simply be:
```
modules.export = {
    ...
    deployProject: callDeployCommand,
    ...
};
```
this way `sfdx.js` still stays a coherent service-like abstraction, but with a profit not enlengthening (it's crazy this word actually exists)  the file with definition of a function doing nothing other than recalling another function with the same arguments.